src/transformState.js

```diff
@@ -6,6 +6,25 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (const singleAction of actions) {
+    switch (singleAction.type) {
+      case 'addProperties':
+        Object.assign(state, singleAction.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of singleAction.keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+    }
```

Comment:
The whole code looks good. One recommandation as per guideline "switch/case should always have default case for error handling."

Reply:
ok thanks for the suggestion. i will add it.